File: tools/repo_root.py
"""Absolute path of this repository, resolved from this file's own location.

Import REPO instead of hardcoding a home directory so the repo works from any
clone path. Override with the IRMSC_ROOT environment variable.
"""
import os
from pathlib import Path

REPO = Path(os.environ.get("IRMSC_ROOT") or Path(__file__).resolve().parent.parent)

File: packages/core/repo-root.mjs
// The absolute path of this repository, resolved from this file's own location.
// Every script imports REPO instead of hardcoding a home directory, so the repo
// works from any clone path. Override with IRMSC_ROOT if you relocate the data.
import { fileURLToPath } from 'node:url';
import { dirname, resolve } from 'node:path';

export const REPO = process.env.IRMSC_ROOT
  || resolve(dirname(fileURLToPath(import.meta.url)), '..');

export default REPO;
